test/e2e/serial/sched_removal_test.go

```diff
@@ -0,0 +1,149 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	e2efixture "github.com/openshift-kni/numaresources-operator/test/utils/fixture"
+	"github.com/openshift-kni/numaresources-operator/test/utils/nrosched"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+	e2ewait "github.com/openshift-kni/numaresources-operator/test/utils/objects/wait"
+)
+
+var _ = Describe("[test_id:47593][serial][disruptive][Scheduler] scheduler removal on a live cluster", func() {
+	var fxt *e2efixture.Fixture
+	var nroSchedObj *nropv1alpha1.NUMAResourcesScheduler
+	var schedulerName string
+
+	BeforeEach(func() {
+		var err error
+		fxt, err = e2efixture.Setup("e2e-test-sched-remove")
+		Expect(err).ToNot(HaveOccurred(), "unable to setup test fixture")
+
+		nroSchedObj = &nropv1alpha1.NUMAResourcesScheduler{}
+		err = fxt.Client.Get(context.TODO(), client.ObjectKey{Name: nrosched.NROSchedObjectName}, nroSchedObj)
+		Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nrosched.NROSchedObjectName)
+
+		schedulerName = nroSchedObj.Status.SchedulerName
+		Expect(schedulerName).ToNot(BeEmpty(), "cannot autodetect the TAS scheduler name from the cluster")
+
+		nrosched.CheckNROSchedulerAvailable(fxt.Client, nroSchedObj.Name)
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("re-creating the NRO Scheduler object: %s", nroSchedObj.Name))
+
+		nroSched := &nropv1alpha1.NUMAResourcesScheduler{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "NUMAResourcesScheduler",
+				APIVersion: nropv1alpha1.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nroSchedObj.Name,
+			},
+			Spec: nroSchedObj.Spec,
+		}
+
+		err := fxt.Client.Create(context.TODO(), nroSched)
+		Expect(err).NotTo(HaveOccurred())
+
+		nrosched.CheckNROSchedulerAvailable(fxt.Client, nroSchedObj.Name)
+
+		err = e2efixture.Teardown(fxt)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	When("removing the topology aware scheduler from a live cluster", func() {
+		It("should keep existing workloads running", func() {
+			var err error
+
+			dp := createDeploymentSync(fxt, "testdp", schedulerName)
+
+			By(fmt.Sprintf("deleting the NRO Scheduler object: %s", nroSchedObj.Name))
+			err = fxt.Client.Delete(context.TODO(), nroSchedObj)
+			Expect(err).ToNot(HaveOccurred())
+
+			maxStep := 3
+			for step := 0; step < maxStep; step++ {
+				time.Sleep(10 * time.Second)
+
+				By(fmt.Sprintf("ensuring the deployment %q keep being ready %d/%d", dp.Name, step, maxStep))
+
+				updatedDp := &appsv1.Deployment{}
+				err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), updatedDp)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(e2ewait.IsDeploymentComplete(dp, &updatedDp.Status)).To(BeTrue(), "deployment %q become unready", dp.Name)
+			}
+		})
+
+		It("should keep new scheduled workloads pending", func() {
```

Comment:
can we recreate the scheduler after this step then verify the pending pods are scheduled?
I am aware that this step wasn't initially in the TC, but I thought it would be useful to check it along with the scenario. another option would be having a separate TC covering this check. wdyt?

Reply:
yes, it's a reasonnable thing to test. Will add a new test case.

Reply:
Thank you. I have created the TC & the id is 48069.

Reply:
thanks, done.